gallery/plot_transforms_v2_e2e.py

```diff
@@ -0,0 +1,147 @@
+"""
+==================================================
+transforms v2: End-to-end object detection example
+==================================================
+"""
+
+import pathlib
+from collections import defaultdict
+
+import PIL.Image
+
+import torch
+import torch.utils.data
+
+import torchvision
+
+
+def show(sample):
+    import matplotlib.pyplot as plt
+
+    from torchvision.transforms.v2 import functional as F
+    from torchvision.utils import draw_bounding_boxes
+
+    image, target = sample
+    if isinstance(image, PIL.Image.Image):
+        image = F.to_image_tensor(image)
+    image = F.convert_dtype(image, torch.uint8)
+    annotated_image = draw_bounding_boxes(image, target["boxes"], colors="yellow", width=3)
+
+    fig, ax = plt.subplots()
+    ax.imshow(annotated_image.permute(1, 2, 0).numpy())
+    ax.set(xticklabels=[], yticklabels=[], xticks=[], yticks=[])
+    fig.tight_layout()
+
+    fig.show()
+
+
+########################################################################################################################
+# To use ``transforms.v2``, you can use the stable datasets and models and just have import transforms differently
+
+# We are using BETA APIs, so we deactivate the associated warning, thereby acknowledging that
+# some APIs may slightly change in the future
+torchvision.disable_beta_transforms_warning()
```

Comment:
Looking at this more we could consider showing the warning (which would have links to the relevant feedback issues) and just mention that one can call `disable_beta_transforms_warning()` to disable it, but no strong opinion.

Reply:
I can push a commit that shows this to have a better base for discussion. Will update here later.

Reply:
See 389103ffe293a880b69bca6db19383c4a5e9dd13

Reply:
Due to

https://github.com/pytorch/vision/blob/a46d97c96dfb2f7f9ddc7f4f889d9856b46428ad/docs/source/conf.py#L36

the warning is not shown even if we don't call that explicitly ourselves in the gallery again:

https://output.circle-artifacts.com/output/job/f238016d-ff53-4d71-ae1d-769bef342b86/artifacts/0/docs/auto_examples/plot_transforms_v2_e2e.html#sphx-glr-auto-examples-plot-transforms-v2-e2e-py

Reply:
Shall we merge the imports into the other ones in this case?

Reply:
> Shall we merge the imports into the other ones in this case?

Not sure what you mean but I'd still rather keep the call to ` torchvision.disable_beta_transforms_warning() ` and keep it close to the v2 imports. Even if it's a no-op in our doc build, it is necessary for users executing this code locally either through copy/pasting, by downloading the associated notebook, or simply executing the `.py` script directly (which is still one of the intended use-cases of the gallery)

Reply:
Ok, let's go back to disabling it explicitly and including a comment.